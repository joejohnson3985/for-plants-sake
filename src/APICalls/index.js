import { hpapikey, gmapikey } from '../apikey.js';


export const getTrails = filters => {
  const { lat, lng, maxDistance, minStars, maxResults, sort, minLength } = filters
  const base = 'https://www.hikingproject.com/data/get-trails?'
  const url = `${base}lat=${lat}&lon=${lng}&maxDistance=${maxDistance}&key=${hpapikey}&minStars=${minStars}&maxResults=${maxResults}&sort=${sort}&minLength=${minLength}`
  return fetch(url).then(response => {
    if (!response.ok) {
      throw new Error('Unable to fetch trails.');
    }
    return response.json();
  })
};

export const getSearchedTrails = address => {
  const url = `https://maps.googleapis.com/maps/api/geocode/json?address=${address}&key=${gmapikey}`
  return fetch(url).then(response => {
    if (!response.ok) {
      throw new Error('Error grabbing that location.');
    }
    return response.json();
  })
}

export const getCurrentLocationName = location => {
  const { lat, lng } = location
  const url = `https://maps.googleapis.com/maps/api/geocode/json?latlng=${lat},${lng}&key=${gmapikey}`
  return fetch(url).then(response => {
    if (!response.ok) {
      throw new Error('Unable to name this location.');
    }
    return response.json();
  })
}

export const getMyHikes = (ids) => {
  const base = 'https://www.hikingproject.com/data/get-trails-by-id?'
  const url = `${base}ids=${ids}&key=${hpapikey}`
  return fetch(url).then(response => {
    if(!response.ok) {
      throw new Error('Error fetching your hikes.')
    }
    return response.json()
  })
}  

