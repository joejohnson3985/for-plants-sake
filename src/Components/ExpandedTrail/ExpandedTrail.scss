@import '../../Utilities/_Variables.scss';

.expanded-trail {
  background: $text-bg;
  width: 85%;
  height: 60%;
  z-index: 6;
  display: flex;
  box-shadow: $heavy-card-shadow;
  border-radius: 4px;

  p {
    color: $dark-text;
  }

}

.trail-overlay{
  @include popup
  background-color: $text-bg;
  height: 90vh;
}

.img-overlay {
  background-color: $text-bg;
  height: 65%;
  width: 90%;
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  border-radius: 8px;

  h1 {
    color: $dark-text;
  }
}

.main-img {
  height: 100%;
  width: 40%;
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  display: flex;
  align-items: center;
  justify-content: center;
}

.widget-ratings {
  width: 100%;
  text-align: center;
} 

.trail-description {
  padding: 10px;
  width: 60%;
  display: flex;
  justify-content: space-between;
  flex-direction: column;

  article {
    width: 100%;
  }
}

.basic-info {
  display: flex;
  align-items:center;
  justify-content: space-between;
} 

.condition-elevation {
  display: flex;
  align-items: center;
  justify-content: space-between;

  div {
    width: 50%;
    height: 100%;
    text-align: center;
    margin: 0 10px;
    border: 1px solid $dark-accent;
    padding: 5px;

    p {
      color: $dark-text;
      margin: 10px 0;
    }

    h1 {
      text-align: center;
      color: $dark-accent;
      margin: 0;
      margin-bottom: 10px;
    }
  }
}

.conditions p {
  text-align: left;
}

.buttons-row {
  display: flex;
  justify-content: space-between;
  align-items: center;

  button{
    color: $dark-text;

    &:hover {
      cursor: pointer;
      box-shadow: $heavy-card-shadow;
      color: black;
    }
  }
}


