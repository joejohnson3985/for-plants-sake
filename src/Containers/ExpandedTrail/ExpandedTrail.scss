@import '../../Utilities/_Variables.scss';

.expanded-trail {
  background: $text-bg;
  width: 85%;
  height: 70%;
  z-index: 6;
  display: flex;
  box-shadow: $heavy-card-shadow;
  border-radius: 4px;

}

.trail-overlay{
  @include popup
  background-color: $text-bg;
  height: 90vh;
}

.img-overlay {
  background-color: $text-bg;
  height: 65%;
  width: 90%;
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  border-radius: 8px;
}

.main-img {
  height: 100%;
  width: 50%;
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  display: flex;
  align-items: center;
  justify-content: center;
}

.widget-ratings {
  width: 100%;
  text-align: center;
} 

.trail-description {
  padding: 10px;
  width: 50%;
  display: flex;
  justify-content: space-between;
  flex-direction: column;

  article {
    width: 100%;
  }
}

.basic-info {
  display: flex;
  align-items:center;
  justify-content: space-between;
} 

.condition-elevation {
  display: flex;

  div {
    width: 50%;
    text-align: center;
  }
}