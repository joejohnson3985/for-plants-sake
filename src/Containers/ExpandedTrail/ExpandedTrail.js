import React, { Component } from 'react';
// import Ratings from 'react-ratings-declarative';
// import noPhoto from '../../Media/no-photo.png';
import './ExpandedTrail.scss'
import { getMyHikes} from '../../APICalls'

class ExpandedTrail extends Component {
  constructor() {
    super()
   this.state = {trail: {}}
  }

  componentDidMount() {
    this.getAllTrailIfno(this.props.match.params.id)
  }

  handleClick = (e) => {
    window.history.back()
  }

  getAllTrailIfno = (id) => {
    getMyHikes(id)
    .then(result => this.setState({trail: result.trails[0]}))
  }  

  render() {
    const { name, difficulty, stars, starVotes, length, imgMedium, expandTrail } = this.state.trail
    return (
      <div className='trail-overlay'>
        <div className='expanded-trail'>
          <p>{name}</p>
          <p>{stars}</p>
          <p>{difficulty}</p>
          <p>{starVotes}</p>
          <p>{length}</p>
          <img src={imgMedium}/>
          <button onClick={() => this.handleClick()}>x</button>
        </div>
      </div>
    )
  }
}

export default ExpandedTrail;