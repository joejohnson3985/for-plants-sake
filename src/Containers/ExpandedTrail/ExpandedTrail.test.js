import React from 'react';
import { getMyHikes} from '../../APICalls'
import { shallow } from 'enzyme';
import {ExpandedTrail} from './ExpandedTrail';

describe('ExpandedTrail', () => {
  let wrapper;
  let expected;
  let mockMatch = {params:{id: 7000130}}

  beforeEach(() => {
    wrapper= shallow(
      <ExpandedTrail match={mockMatch}/>
    )
  })

  describe('Component', () => {

    it('Should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    })

    it('should have a default state', () => {
      expected = {trail: {}}
      expect(wrapper.state()).toEqual(expected)
    })
  })
})