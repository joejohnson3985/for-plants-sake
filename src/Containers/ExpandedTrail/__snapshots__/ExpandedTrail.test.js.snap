// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpandedTrail Component Should match the snapshot 1`] = `
<div
  className="trail-overlay"
>
  <div
    className="expanded-trail"
  >
    <section
      className="main-img"
      style={
        Object {
          "backgroundImage": "url(no-photo.png)",
        }
      }
    >
      <article
        className="img-overlay"
      >
        <h1 />
        <Ratings
          changeRating={null}
          gradientPathName=""
          ignoreInlineStyles={false}
          rating={0}
          svgIconPaths="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
          svgIconViewBoxes="0 0 51 48"
          typeOfWidget="Star"
          widgetDimensions="50px"
          widgetEmptyColors="rgb(203, 211, 227)"
          widgetHoverColors="rgb(230, 67, 47)"
          widgetRatedColors="#df7975"
          widgetSpacings="7px"
        >
          <Widget
            widgetDimension="25px"
          />
          <Widget
            widgetDimension="25px"
          />
          <Widget
            widgetDimension="25px"
          />
          <Widget
            widgetDimension="25px"
          />
          <Widget
            widgetDimension="25px"
          />
        </Ratings>
      </article>
    </section>
    <section
      className="trail-description"
    >
      <article
        className="basic-info"
      >
        <img
          src="https://cdn.apstatic.com/img/diff/undefined.svg"
        />
        <p>
           stars from 
           votes
        </p>
        <p>
           miles
        </p>
      </article>
      <article
        className="condition-elevation"
      >
        <div>
          <h1>
            Conditions
          </h1>
          <table>
            <tbody>
              <tr>
                <td>
                  Condition:
                </td>
                <td />
              </tr>
              <tr>
                <td>
                  Details:
                </td>
                <td />
              </tr>
              <tr>
                <td>
                  Date:
                </td>
                <td>
                  .
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div>
          <h1>
            Elevation
          </h1>
          <table>
            <tbody>
              <tr>
                <td>
                  Ascent
                </td>
                <td>
                   ft.
                </td>
              </tr>
              <tr>
                <td>
                  Descent:
                </td>
                <td>
                   ft.
                </td>
              </tr>
              <tr>
                <td>
                  Highest Point:
                </td>
                <td>
                   ft.
                </td>
              </tr>
              <tr>
                <td>
                  Lowest Point:
                </td>
                <td>
                   ft.
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </article>
      <article>
        <p />
      </article>
      <article>
        <a
          rel="noopener noreferrer"
          target="_blank"
        >
          <button>
            REI
          </button>
        </a>
        <button
          onClick={[Function]}
        >
          x
        </button>
      </article>
    </section>
  </div>
</div>
`;
